lib/rex/post/smb/ui/console/command_dispatcher/shares.rb

```diff
@@ -318,7 +344,198 @@ def cmd_cat(*args)
           def cmd_cd_tabs(_str, words)
             return [] if words.length > 1
 
-            @@cat_opts.option_keys
+            @@cd_opts.option_keys
+          end
+
+          def cmd_upload(*args)
+            if args.include?('-h') || args.include?('--help')
+              cmd_upload_help
+              return
+            end
+            local_path = nil
+            remote_path = nil
+
+            @@upload_opts.parse(args) do |_opt, idx, val|
+              case idx
+              when 0
+                local_path = val
+              when 1
+                remote_path = val
+              else
+                print_warning('Too many parameters')
+                cmd_upload_help
+              end
+            end
+
+            if local_path.blank?
+              print_error('No local path given')
+              return
+            end
+
+            remote_path = Rex::Post::File.basename(local_path) if remote_path.nil?
+            full_path = as_ntpath(Pathname.new(shell.cwd).join(remote_path).to_s)
+
+            fd = simple_client.open(full_path, 'wct', write: true)
+
+            data = ::File.read(local_path, ::File.size(local_path), mode: 'rb')
+            fd.write(data)
+            fd.close
+
+            print_good("#{local_path} uploaded to #{full_path}")
+          end
+
+          def cmd_upload_tabs(str, words)
+            tab_complete_filenames(str, words)
+          end
+
+          def cmd_upload_help
+            print_line 'Usage: upload <local_path> <remote_path>'
+            print_line
+            print_line 'Upload a file to the remote target.'
+            print @@upload_opts.usage
+          end
+
+          def cmd_download(*args)
```

Comment:
Placeholder comment from me incase I forgot; Looks like this doesn't work for me

This branch

```
SMB (127.0.0.1\foo_share) > ls
ls
==

    #  Type  Name  Created                    Accessed                   Written                    Changed                    Size
    -  ----  ----  -------                    --------                   -------                    -------                    ----
    0  DIR   .     2024-02-28T13:39:12+00:00  2024-02-28T13:43:59+00:00  2024-02-28T13:39:12+00:00  2024-02-28T13:39:12+00:00
    1  DIR   ..    2024-02-28T13:45:18+00:00  2024-02-28T13:45:18+00:00  2024-02-28T13:45:25+00:00  2024-02-28T13:45:25+00:00
    2  FILE  bar   2024-02-28T13:39:12+00:00  2024-02-28T13:45:25+00:00  2024-02-28T13:39:12+00:00  2024-02-28T13:39:12+00:00  4

SMB (127.0.0.1\foo_share) > download bar
[-] Error running command download: RubySMB::Error::UnexpectedStatusCode The server responded with an unexpected status code: STATUS_OBJECT_NAME_NOT_FOUND

```

Whilst with smb client:
```
smb: \> get bar  
getting file \bar of size 4 as bar (1.3 KiloBytes/sec) (average 1.3 KiloBytes/sec)
```

Reply:
I tried taking a quick look at wireshark, but it looks like the login scanner doesn't honor the `smb::alwaysencrypt=false` and `smb::protocolversion=2` options - so it's all encrypted by default

Reply:
that's weird, can you `cat` the file?

Reply:
I'll resolve this for now, but I think there's still an issue here